### PR TITLE
Moving the detection of manual conversion functions to a dedicated struct

### PR DESCRIPTION
**Pull request by wk8**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

This PR is the first of a series of PRs aimed at making
https://github.com/kubernetes/kubernetes/pull/83621
easier to review; the final goal is to separate generic conversion logic
from the bits that are specific to Kubernetes, so that other projects
can re-use this library for generating code converting structs from one
package to structs from another.

**Special notes for your reviewer**:

Please see the discussion at https://github.com/kubernetes/kubernetes/pull/83621,
and more specifically https://github.com/kubernetes/kubernetes/pull/83621#issuecomment-548967945

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by lavalamp** [comment]
The `verify` test has some things for you to fix.

otherwise, lgtm

**Comment by wk8** [comment]
Sorry it took me a while to get around to it, ready for another review @lavalamp . Thanks!

File: staging/src/k8s.io/code-generator/pkg/conversiongen/doc.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conversiongen
+
+// This package contains generic structures and helpers to write conversion generators.
```
Comment (lavalamp) [inline staging/src/k8s.io/code-generator/pkg/conversiongen/doc.go]: Doc comment goes on the line(s) immediately before the `package` keyword (to show up in godoc) and has to start with "Package conversiongen ..." (to make go vet happy).

Comment (wk8) [reply staging/src/k8s.io/code-generator/pkg/conversiongen/doc.go]: 👍

File: staging/src/k8s.io/code-generator/pkg/conversiongen/utils.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	"k8s.io/klog"
+)
+
+// ConversionPair represents a conversion pair from inType to outType
+type ConversionPair struct {
+	InType  *types.Type
+	OutType *types.Type
+}
+
+// A NamedVariable represents a named variable to be rendered in snippets.
```
Comment (lavalamp) [inline staging/src/k8s.io/code-generator/pkg/conversiongen/utils.go]: Hm, gengo probably needs to get this concept at some point (not asking for a change).

**Comment by lavalamp** [comment]
/lgtm
/approve

Sorry for the delay, I was OOO

**Comment by wk8** [comment]
Thanks for the review @lavalamp , will ping you on subsequent PRs :)

**Comment by wk8** [comment]
@lavalamp : had to do a typo change, could you please LGTM again? Thank you!

**Pull request closed without merging**